### PR TITLE
feat: adopt hardhat for development

### PR DESCRIPTION
**Pull request by mirceanis**
I also refactored the tests and added some extra ones to cover some corner cases for the `doSomethingSigned()` methods.

### CODE REVIEW

File: scripts/generateDeployTxs.js [hunk 1]
```diff
@@ -1,55 +1,57 @@
...
 const gasLimits = {
   EthereumDIDRegistry: 2811144, // If this value needs to be recalculated, it can be done by deploying the rawTx once and looking at gasUsed in the receipt
 }
 
-generateDeployTx = (code, name) => {
+const generateDeployTx = (code, name) => {
   const rawTx = {
     nonce: 0,
-    gasPrice: 10000000000, // 10 Gwei
+    gasPrice: 100000000000, // 100 Gwei
```
Comment (mirceanis) [inline scripts/generateDeployTxs.js:16]: modern times, modern prices
